di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/utils/InheritedIdentityJWTBuilder.java

```diff
@@ -0,0 +1,102 @@
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECPrivateKey;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_ISSUER;
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_SIGNING_KEY;
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_TTL;
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_VTM;
+
+public class InheritedIdentityJWTBuilder {
+
+    public static final String VC_CREDENTIAL_SUBJECT = "credentialSubject";
+    public static final String VOT = "vot";
+    public static final String VTM = "vtm";
+    public static final String VC = "vc";
+    public static final String VC_TYPE = "type";
+    public static final String VERIFIABLE_CREDENTIAL_TYPE = "VerifiableCredential";
+    public static final String INHERITED_IDENTITY_CREDENTIAL_TYPE = "InheritedIdentityCredential";
+    public static final String VC_EVIDENCE = "evidence";
+
+    public static final String EC_ALGO = "EC";
+
+    public static SignedJWT generate(
+            String userId, String vtr, String credentialSubject, String evidence)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+        Map<String, Object> vc = new LinkedHashMap<>();
+        vc.put(
+                VC_TYPE,
+                new String[] {VERIFIABLE_CREDENTIAL_TYPE, INHERITED_IDENTITY_CREDENTIAL_TYPE});
+        vc.put(VC_CREDENTIAL_SUBJECT, convertJsonToMap(credentialSubject));
+        vc.put(VC_EVIDENCE, List.of(convertJsonToMap(evidence)));
+        Instant now = Instant.now();
+        JWTClaimsSet claimsSet =
+                new JWTClaimsSet.Builder()
+                        .subject(userId)
+                        .issuer(INHERITED_IDENTITY_JWT_ISSUER)
+                        .notBeforeTime(Date.from(now))
+                        .expirationTime(generateExpirationTime(now))
+                        .claim(VOT, vtr)
+                        .claim(VTM, INHERITED_IDENTITY_JWT_VTM)
+                        .claim(VC, vc)
+                        .build();
+        return createSignedJwt(claimsSet);
+    }
+
+    private static Map<String, Object> convertJsonToMap(String json) {
+        try {
+            return new ObjectMapper().readValue(json, new TypeReference<>() {});
```

Comment:
Might be worth making a static object mapper that can be reused rather than generating a new one very time.